src/rtc_peer.ts

```diff
@@ -193,33 +351,48 @@ export class RTCPeer extends EventEmitter {
         const start = performance.now();
 
         return new Promise<void>((resolve, reject) => {
-            this.dcLockResponseCb = (acquired) => {
-                if (acquired) {
-                    this.logger.logDebug(`RTCPeer.grabSignalingLock: lock acquired in ${Math.round(performance.now() - start)}ms`);
-                    this.dcLockResponseCb = null;
-                    resolve();
-                    return;
+            // The attemptLock wrapper is needed since Promise executor should be synchronous.
+            const attemptLock = async () => {
+                // This covers the case of "concurrent" (interleaved in practice) attempts to lock
+                // which would otherwise result in this.dcLockResponseCb getting overwritten.
+                // Waiting ensures lock attempts are all done fully serially.
```

Comment:
I haven't review current code so I don't think this case is easy to reproduce but it's something we should probably port to master sooner since it's a bug :)